Tensile/TensileRetuneLibrary.py

```diff
@@ -0,0 +1,178 @@
+# Copyright 2016-2021 Advanced Micro Devices, Inc. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell cop-
+# ies of the Software, and to permit persons to whom the Software is furnished
+# to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IM-
+# PLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+# FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNE-
+# CTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+################################################################################
+
+from . import BenchmarkProblems
+from . import ClientExecutable
+from . import ClientWriter
+from . import LibraryIO
+from .Common import globalParameters, print1, printExit, ensurePath, assignGlobalParameters, \
+                    pushWorkingPath, popWorkingPath, restoreDefaultGlobalParameters, HR
+from .Tensile import addCommonArguments
+from .SolutionStructs import ProblemSizes
+from . import __version__
+
+import argparse
+import copy
+import os
+import shutil
+import sys
+
+
+def parseCurrentLibrary(libPath):
+    global globalParameters
+
+    libYaml = LibraryIO.readYAML(libPath)
+    # parseLibraryLogicData mutates the original data, so make a copy
+    fields = LibraryIO.parseLibraryLogicData(copy.deepcopy(libYaml), libPath)
+    (_, _, problemType, solutions, _, exactLogic, _, _, _) = fields
+
+    # get performance metric
+    if len(libYaml) > 10:
+        globalParameters["PerformanceMetric"] = libYaml[10]
+
+    # process exactLogic into ProblemSizes
+    sizes = []
+    for (size, _) in exactLogic:
+        sizes.append({"Exact": size})
+    problemSizes = ProblemSizes(problemType, sizes)
+
+    return (libYaml, solutions, problemSizes)
+
+
+def runBenchmarking(solutions, problemSizes, outPath):
+    # TODO some copy-pasting from BenchmarkProblems.benchmarkProblemType
+    # could use a refactor to elimate duplicated code
+    global globalParameters
+
+    ClientExecutable.getClientExecutable()
+
+    shortName = "benchmark"
+    benchmarkDir = os.path.join(outPath, shortName)
+    sourceDir = os.path.join(benchmarkDir, "source")
+    ensurePath(sourceDir)
+
+    pushWorkingPath(shortName)
+    pushWorkingPath("source")
+
+    filesToCopy = [
```

Comment:
I think we can omit the old client stuff from here.

Reply:
I agree in theory, but there appears to be a bug with "Kernels.h" and "Kernels.cpp" being generated and then not compiling due to missing includes if you skip these steps. I think I noticed a similar thing a while ago with the normal Tensile client trying to run it with OldClientSourceTmp = False. 

I'll take a look at fixing the underlying issue on Monday. If it ends up being easy to fix, I'll update this. Otherwise, I'll leave this hacky fix for now so we can start retuning and then reevaluate after that's done.